src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/integration/SetupAssistant.kt

```diff
@@ -539,31 +539,32 @@ class SetupAssistant(
     referral.accessibilityNeeds = accessibilityNeeds
     referral.additionalNeedsInformation = additionalNeedsInformation
     referral.additionalRiskInformation = additionalRiskInformation
-    referral.completionDeadline = completionDeadline
     referral.complexityLevelIds = complexityLevelIds
-    referral.furtherInformation = furtherInformation
     referral.hasAdditionalResponsibilities = hasAdditionalResponsibilities
     referral.interpreterLanguage = interpreterLanguage
-    referral.maximumEnforceableDays = maximumEnforceableDays
     referral.needsInterpreter = needsInterpreter
     referral.relevantSentenceId = relevantSentenceId
     referral.whenUnavailable = whenUnavailable
 
-    referralDetailsRepository.save(
-      ReferralDetails(
-        UUID.randomUUID(),
-        null,
-        referral.id,
-        referral.createdAt,
-        referral.createdBy.id,
-        "initial referral details",
-        completionDeadline,
-        furtherInformation,
-        maximumEnforceableDays,
+    return referralRepository.save(referral).also {
+      val details = referralDetailsRepository.save(
```

Comment:
Given this is a setup tool, do we need to care about these executing in the same transaction?

Reply:
yeh unfortunately the order in which these things gets done is important to avoid sql constraint violations